avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroTypeDeserializer.java

```diff
@@ -57,6 +54,13 @@ public Object deserializeTypedFromScalar(JsonParser p, DeserializationContext ct
 
     @Override
     public Object deserializeTypedFromAny(JsonParser p, DeserializationContext ctxt) throws IOException {
+        AnnotationIntrospector ai = ctxt.getAnnotationIntrospector();
```

Comment:
I feel like instead of having the type deserializer look for custom deserializers, there should be some way to avoid using the type deserializer at all for properties that have custom deserializers - but I don't know where that check needs to be made.